Add Track Events to documentation links.

tomalec opened the pull request:
:warning:  This PR is based on https://github.com/woocommerce/pinterest-for-woocommerce/pull/254, for better documentation 
<!-- You can erase any parts of this template not applicable to your pull request. -->

<!-- Reference issue(s) that this PR fixes or addresses -->
Implements part of #232 

### Changes proposed in this pull request
<!-- Explain what this PR adds or changes, why, and how this will impact users. -->
Add Track Events to documentation links.

Change `StepOverview` to accept more props for read more `Button` link.

#### Screenshots
<!--- Optional --->

### Detailed test instructions
<!-- Add steps to confirm the fix or change. -->
1. Enable tracks logging, by `localStorage.setItem('debug', 'wc-admin*')`
2. Check the `wcadmin_pfw_documentation_link_click` events fired for the following links:
	- at [`/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Flanding`](https://wp581wc580.test/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Flanding)
		-  "Terms of Service." - `{ link_id: 'terms-of-service', context: 'welcome-section' }`
	- at [`/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Fonboarding`](https://wp581wc580.test/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Fonboarding)
		- "Pinterest advertising guidelines." - `{ link_id: 'ad-guidelines', context: 'setup-account' }`
		![image](https://user-images.githubusercontent.com/17435/141852692-c06974ec-596e-4ee2-9036-86d6d40e3202.png)



	- at [`/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Fonboarding&step=claim-website`](https://wp581wc580.test/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Fonboarding&step=claim-website)
		- "Read more" - `{ link_id: 'claim-website', context: 'claim-website' }` 
		![image](https://user-images.githubusercontent.com/17435/141852627-3e03cbc6-6720-4cc9-bd95-50903fca61c9.png)
	- at [`/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Fonboarding&step=setup-tracking`](https://wp581wc580.test/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Fonboarding&step=setup-tracking)
		-  "Ad Guidelines" - `{ link_id: 'ad-guidelines', context: 'wizard'|'settings' }`
		-  "Ad Data Terms" - `{ link_id: 'ad-data-terms', context: 'wizard'|'settings' }`
		-  "Read more" - `{ link_id: 'setup-tracking', context: 'wizard'|'settings' }`
		- "Pinterest Advertising Agreement" - `{ link_id: 'tos', context: 'wizard'|'settings' }`
	![image](https://user-images.githubusercontent.com/17435/141852368-bd797107-39cf-49ba-9b97-87eb4bfeaba3.png)



<!-- Please add details of other areas that might be impacted by the change. -->

(no changelog, as we will add a single entry for a feature branch)

### Changelog note

tomalec committed:
Add Track Events to documentation links.

Change `StepOverview` to accept more props for read more `Button` link.

eason9487 commented on assets/source/setup-guide/app/steps/SetupAccount.js:
```diff
@@ -137,6 +138,20 @@ const SetupAccount = ( {
 										}
 										target="_blank"
 										rel="noreferrer"
+										onClick={ () =>
+											recordEvent(
+												'pfw_documentation_link_click',
+												{
+													link_id: 'ad-guidelines',
+													context: 'setup-account',
+													href:
+														wcSettings
+															.pinterest_for_woocommerce
+															.pinterestLinks
+															.adGuidelines,
+												}
+											)
+										}
```
📜 
I think this probably wouldn't conflict with #255, but it added a new link to the description. Perhaps the corresponding event tracking implementation could be added directly to #255 by the way?
// cc @ianlin

tomalec replied:
This PR is targetting the feature branch https://github.com/woocommerce/pinterest-for-woocommerce/tree/feature/232-jsdoc-tweaks/ , as we want to deliver all tracks at once. I guess it would be cleaner to add that event here (or in the following PR) once https://github.com/woocommerce/pinterest-for-woocommerce/tree/feature/210-settings-pinterest-merchant-guidelines is merged to `develop`, then `develop` to here

eason9487 replied:
That's a better and thorough way. 👍  Thanks!

eason9487 commented on assets/source/setup-guide/app/steps/SetupTracking.js:
```diff
@@ -329,10 +373,19 @@ const SetupTracking = ( { view } ) => {
-						}
+						readMore={ {
+							href:
+								wcSettings.pinterest_for_woocommerce
+									.pinterestLinks.SetupTracking,
+							onClick: () =>
+								recordEvent( 'pfw_documentation_link_click', {
+									link_id: 'setup-tracking',
+									context: view,
+									href:
+										wcSettings.pinterest_for_woocommerce
+											.pinterestLinks.SetupTracking,
```
📝
The property `SetupTracking` doesn't exist, and I think it may have been a problem from [the beginning](https://github.com/woocommerce/pinterest-for-woocommerce/commit/b19c441e4591cc623b689bb55ff0c329184b9621#diff-39ee87358882006b4b222b52f6a4cbf74df2e0267b947b808ca7f4c16dc3f08cR73). I will create an issue for it later.

eason9487 replied:
Created #262.

eason9487 commented on assets/source/setup-guide/app/views/LandingPageApp.js:
```diff
@@ -21,6 +21,16 @@ import {
  */
 import PrelaunchNotice from '../../../components/prelaunch-notice';
 
+const tosHref = 'https://business.pinterest.com/business-terms-of-service/';
```
💅 
Looks like Pinterest has an issue when detecting user's browser language. After a few 301 and 302 redirects, it took me to a 404 page - https://business.pinterest.com/zh/zh/zh/zh/business-terms-of-service. Maybe replace it with `'https://business.pinterest.com/en/business-terms-of-service/'`?
 
![Kapture 2021-11-16 at 17 28 24](https://user-images.githubusercontent.com/17420811/141959198-05beb534-93df-4389-9565-e36dcc85840f.gif)

tomalec replied:
I'm able to reproduce it by changing the preferred language in my browser to Simplified Chinese. But https://business.pinterest.com/zh/business-terms-of-service/ works when it's set to English.
Also redirecting works fine for other languages, like Japanese and Polish.

I created a separate issue to track it https://github.com/woocommerce/pinterest-for-woocommerce/issues/264

eason9487 reviewed (approved):
Tested locally and most of all work well. Left three minor non-blocking comments. LGTM.

tomalec committed:
Merge branch 'feature/232-tracking' into feature/232-docs

tomalec committed:
Add Track Event for merchant guidelines link.

After https://github.com/woocommerce/pinterest-for-woocommerce/pull/255 was merged to `feature/232-tracking`

tomalec commented:
@eason9487  Would you mind another look? I added an event for the link added in `develop` by #255. The other comments have now separate issues.

I also created a DRY improvement at https://github.com/woocommerce/pinterest-for-woocommerce/pull/266, but I'm not sure if it's correct in respect of React hooks. Also, I'm not sure what we prefer in terms of balance between DRY code base and explicitly of Track Events being fired.

eason9487 reviewed (approved):
> I added an event for the link added in `develop` by #255.

The added commit looks good and works as expected.
